spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -85,6 +117,19 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
       throw new IllegalArgumentException("Sampling fraction must be larger or equal to 0 in sampling with replacement")
     }
 
+    getRandomSampleByNum((fraction * mDDF.getNumRows).toLong, withReplacement, seed)
+
+  }
+
+  override def getRandomSampleApprox(fraction: Double, withReplacement: Boolean, seed: Int): DDF = {
+    if (!withReplacement && (fraction > 1 || fraction < 0)) {
```

Comment:
humm, empty DDF is still a valid one, right? Here is from Spark's sample doc https://spark.apache.org/docs/1.6.0/api/java/org/apache/spark/rdd/RDD.html#sample(boolean,%20double,%20long)